File: tournaments/is-admissible-overpayment/is-admissible-over-payment.py
def isAdmissibleOverpayment(P, N, X):
    t = 0
    for p, n in zip(P, N):
        d = 0
        s = n.split('%')
        if s[1:]:
            d = float(s[0]) / 100
            if 'lo' in n:
                d *= -1
        d += 1
        t += round(p - p / d, 8);
    return t <= X


# notes = ["20.00% lower than in-store",  "10.00% higher than in-store"]
# prices = [48, 165]
# x = 2

prices = [110, 95, 70]
notes = ["10.0% higher than in-store", "5.0% lower than in-store", "Same as in-store"]
x = 5
# prices = [40, 40, 40, 40]
# notes = ["0.001% higher than in-store", "0.0% lower than in-store", "0.0% higher than in-store", "0.0% lower than in-store"]
# x = 0
print(isAdmissibleOverpayment(prices, notes, x))

File: tournaments/is-admissible-overpayment/is-admissible-overpayment.js
function isAdmissibleOverpayment(prices, notes, x) {
    let difference = 0;
    
    for (i = 0; i < prices.length; i++) {
        const price = prices[i];
        const note = notes[i];
        const words = note.split(' ');
        
        if (words[0] === 'Same') { continue; }
            
        const first = words[0];
        const percent = Number(first.slice(0, first.length - 1)) / 100;
        let inStoreCost;
        
        if (words[1] === 'higher') {
            inStoreCost =  price / (1 + percent);
        }
        
        if (words[1] === 'lower') {
            inStoreCost =  price / (1 - percent);
        }
        
        difference += price - inStoreCost;
    }
    
    console.log(difference);
    return Math.round(difference * 1000000) <= x * 1000000;
}

function isAdmissibleOverpaymentV2(prices, notes, x) {
    var diff = 0,
        i,
        price, note, orig;
    for (i in prices) {
        price = prices[i];
        note = notes[i].match(/(.*)% (lower|higher)/);
        if (note) {
            if (note[2] == 'higher') {
                orig = price * 100 / (100 - - note[1]);
            } else {
                orig = price * 100 / (100 - note[1]);
            }
        } else {
            orig = price;
        }
        diff += price - Math.round(orig * 1e6) / 1e6;
    }
    return x >= diff;
}